cli/tsc/dts/lib.es2021.promise.d.ts

```diff
@@ -45,4 +45,10 @@ interface PromiseConstructor {
      * @returns A new Promise.
      */
     any<T>(values: Iterable<T | PromiseLike<T>>): Promise<Awaited<T>>;
+
+    /**
+     * Creates a Promise that can be resolved or rejected using provided functions.
+     * @returns An object containing `promise` promise object, `resolve` and `reject` functions.
+     */
+    withResolvers<T>(): { promise: Promise<T>; resolve: (value: T | PromiseLike<T>) => void; reject: (reason?: any) => void; };
```

Comment:
This feature should be ES2024

Reply:
As of TypeScript 5.5.2, it seems to be defined in [lib.esnext.promise.d.ts](https://github.com/microsoft/TypeScript/blob/v5.5.2/lib/lib.esnext.promise.d.ts). 

I work in progress to move this to lib.es2024.promise.d.ts.
https://github.com/microsoft/TypeScript/pull/58573